projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -179,9 +179,22 @@ public static function admin_init() {
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- No nonce needed for redirect flow control
 		$step = isset( $_GET['step'] ) ? sanitize_text_field( wp_unslash( $_GET['step'] ) ) : '';
 
-		// Handle onboarding redirects based on connection status
+		$redirect_args = array( 'page' => 'my-jetpack' );
+
+		if ( 'connect-user' === $step ) {
```

Comment:
In my opinion, this obscures the logic. Instead of JS handling routes, it's intercepted by PHP. I believe this will be harder to debug in a few years.
I won't block this if the team wants to implement it this way, but I prefer `handleConnectUser` from `useMyJetpackConnection` because it utilises an existing hook. The proposed implementation adds a new route redirect.

Reply:
I deliberately placed the logic where we handle the other `step` (i.e. `onboarding`) to make it easier to find.

Reply:
> I prefer `handleConnectUser` from `useMyJetpackConnection` because it utilises an existing hook. The proposed implementation adds a new route redirect.

That is not possible in all the cases as mentioned in https://github.com/Automattic/jetpack/pull/44469#issuecomment-3130476347